tools/crm_mon.c

```diff
@@ -2994,60 +2994,41 @@ print_failed_actions(FILE *stream, pe_working_set_t *data_set)
 static stonith_history_t *
 reduce_stonith_history(stonith_history_t *history)
 {
-    stonith_history_t *new = NULL, *hp, *np, *tmp;
+    stonith_history_t *new = NULL, *hp, *np;
 
     for (hp = history; hp; ) {
-        for (np = new; np; np = np->next) {
-            if ((hp->state == st_done) || (hp->state == st_failed)) {
-                /* action not in progress */
-                if (safe_str_eq(hp->target, np->target) &&
-                    safe_str_eq(hp->action, np->action) &&
-                    (hp->state == np->state)) {
-                    if ((hp->state == st_done) ||
-                        safe_str_eq(hp->delegate, np->delegate)) {
-                        /* replace or purge */
-                        if (hp->completed < np->completed) {
+        stonith_history_t *hp_next = hp->next;
+
+        hp->next = NULL;
+        if (!new) {
+            new = hp;
```

Comment:
Not a problem, but this same effect could be achieved by setting new = history before the loop, and starting the loop with hp = new->next

Reply:
Your opinion is correct.
We'll fix that as it gives a better view of the loop.